src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp

```diff
@@ -25,131 +18,72 @@ HOSTFXR_UTILITY::~HOSTFXR_UTILITY()
-    }
+    const fs::path physicalPath(pcwzApplicationPhysicalPath);
+    const fs::path hostFxrLocation = physicalPath / L"hostfxr.dll";
 
-    if (!UTILITY::CheckIfFileExists(struHostFxrPath.QueryStr()))
+    if (!exists(hostFxrLocation))
     {
-        // Most likely a full framework app.
-        // Check that the runtime config file doesn't exist in the folder as another heuristic.
-        if (FAILED(hr = struRuntimeConfigLocation.Copy(struDllPath)) ||
-            FAILED(hr = struRuntimeConfigLocation.Append(L".runtimeconfig.json")))
-        {
-            goto Finished;
-        }
-        if (!UTILITY::CheckIfFileExists(struRuntimeConfigLocation.QueryStr()))
-        {
+        fs::path runtimeConfigLocation = exePath;
+        runtimeConfigLocation.replace_extension(L".runtimeconfig.json");
 
-            hr = E_APPLICATION_ACTIVATION_EXEC_FAILURE;
-            UTILITY::LogEventF(hEventLog,
-                EVENTLOG_ERROR_TYPE,
-                ASPNETCORE_EVENT_INPROCESS_FULL_FRAMEWORK_APP,
-                ASPNETCORE_EVENT_INPROCESS_FULL_FRAMEWORK_APP_MSG,
-                pcwzApplicationPhysicalPath,
-                hr);
-        }
-        else
+        if (!exists(runtimeConfigLocation))
         {
-            // If a runtime config file does exist, report a file not found on the app.exe
-            hr = HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
-            UTILITY::LogEventF(hEventLog,
-                EVENTLOG_ERROR_TYPE,
-                ASPNETCORE_EVENT_APPLICATION_EXE_NOT_FOUND,
-                ASPNETCORE_EVENT_APPLICATION_EXE_NOT_FOUND_MSG,
-                pcwzApplicationPhysicalPath,
-                hr);
+            EVENTLOG(hEventLog, INPROCESS_FULL_FRAMEWORK_APP, pcwzApplicationPhysicalPath, 0);
+            return E_FAIL;
         }
 
-        goto Finished;
-    }
-
-    if (FAILED(hr = pStruHostFxrDllLocation->Copy(struHostFxrPath)))
-    {
-        goto Finished;
+        EVENTLOG(hEventLog, APPLICATION_EXE_NOT_FOUND, pcwzApplicationPhysicalPath, 0);
+        return E_FAIL;
     }
 
+    fs::path dllPath = exePath;
+    dllPath.replace_extension(".dll");
 
-    if (FAILED(hr = struDllPath.Append(L".dll")))
+    if (!exists(dllPath))
     {
-        goto Finished;
+        return E_FAIL;
     }
 
-    if (!UTILITY::CheckIfFileExists(struDllPath.QueryStr()))
-    {
-        // Treat access issue as File not found
-        hr = HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
-        goto Finished;
-    }
+    auto arguments = std::wstring(dllPath) + L" " + pcwzArguments;
 
-    if (FAILED(hr = struArguments.Copy(struDllPath)) ||
-        FAILED(hr = struArguments.Append(L" ")) ||
-        FAILED(hr = struArguments.Append(pcwzArguments)))
+    if (FAILED(hr = pStruHostFxrDllLocation->Copy(hostFxrLocation.c_str())))
     {
-        goto Finished;
+        return hr;
     }
 
-    if (FAILED(hr = ParseHostfxrArguments(
-        struArguments.QueryStr(),
+    return ParseHostfxrArguments(
+        arguments.c_str(),
```

Comment:
Just make this accept a wstring. Extra copy.

Reply:
Didn't want to change consumers yet.